panther_description/launch/overwrite_robot_description.launch.py

```diff
@@ -0,0 +1,178 @@
+        default_value=EnvironmentVariable("ROBOT_NAMESPACE", default_value=""),
+        description="Add namespace to all launched nodes.",
+    )
+
+    use_sim = LaunchConfiguration("use_sim")
+    declare_use_sim_arg = DeclareLaunchArgument(
+        "use_sim",
+        default_value="False",
+        description="Whether simulation is used.",
+        choices=["True", "true", "False", "false"],
+    )
+
+    wheel_config_path = LaunchConfiguration("wheel_config_path")
+    declare_wheel_config_path_arg = DeclareLaunchArgument(
+        "wheel_config_path",
+        default_value=PathJoinSubstitution(
+            [
+                FindPackageShare("panther_description"),
+                "config",
+                PythonExpression(["'", wheel_type, ".yaml'"]),
+            ]
+        ),
+        description=(
+            "Path to wheel configuration file. By default, it is located in "
+            "'panther_description/config/{wheel_type}.yaml'. You can also specify the path "
+            "to your custom wheel configuration file here. "
+        ),
+    )
+
+    declare_wheel_type_arg = DeclareLaunchArgument(
+        "wheel_type",
+        default_value="WH01",
+        description=(
+            "Specify the wheel type. If the selected wheel type is not 'custom', "
+            "the 'wheel_config_path' and 'controller_config_path' arguments will be "
+            "automatically adjusted and can be omitted."
+        ),
+        choices=["WH01", "WH02", "WH04", "custom"],
+    )
+
+    # Get URDF via xacro
+    imu_pos_x = os.environ.get("ROBOT_IMU_LOCALIZATION_X", "0.168")
+    imu_pos_y = os.environ.get("ROBOT_IMU_LOCALIZATION_Y", "0.028")
+    imu_pos_z = os.environ.get("ROBOT_IMU_LOCALIZATION_Z", "0.083")
+    imu_rot_r = os.environ.get("ROBOT_IMU_ORIENTATION_R", "3.14")
+    imu_rot_p = os.environ.get("ROBOT_IMU_ORIENTATION_P", "-1.57")
+    imu_rot_y = os.environ.get("ROBOT_IMU_ORIENTATION_Y", "0.0")
+    robot_description_content = Command(
+        [
+            PathJoinSubstitution([FindExecutable(name="xacro")]),
+            " ",
+            PathJoinSubstitution(
+                [FindPackageShare("panther_description"), "urdf", "panther.urdf.xacro"]
+            ),
+            " use_sim:=",
+            use_sim,
+            " wheel_config_file:=",
+            wheel_config_path,
+            " controller_config_file:=",
+            controller_config_path,
+            " battery_config_file:=",
+            battery_config_path,
+            " imu_xyz:=",
+            f"'{imu_pos_x} {imu_pos_y} {imu_pos_z}'",
+            " imu_rpy:=",
+            f"'{imu_rot_r} {imu_rot_p} {imu_rot_y}'",
+            " namespace:=",
+            namespace,
+            " components_config_path:=",
+            components_config_path,
+        ]
+    )
+
+    set_robot_description = ExecuteProcess(
+        cmd=[
+            "ros2",
+            "param",
+            "set",
+            "/panther/robot_state_publisher",
```

Comment:
The node name should  be depend on the namespace
```suggestion
            [namespace, "/robot_state_publisher"],
```